src/cljc/athens/common_events/resolver.cljc

```diff
@@ -861,6 +861,43 @@
     tx-data))
 
 
+(defmethod resolve-event-to-tx :datascript/selected-delete
+  [db {:event/keys [args]}]
+  ;; We know that we only need to dec indices after the last block. The former blocks
+  ;; are necessarily going to remove all tail children, meaning we only need to be
+  ;; concerned with the last N blocks that are selected, adjacent siblings, to
+  ;; determine the minus-after value.
+  (println "resolver :datascript/drop-multi-child args" (pr-str args))
+  (let [{:keys [uids]}                    args
+        selected-sibs-of-last             (->> uids
+                                               (d/q '[:find ?sib-uid ?o
+                                                      :in $ ?uid [?selected ...]
+                                                      :where
+                                                      ;; get all siblings of the last block
+                                                      [?e :block/uid ?uid]
+                                                      [?p :block/children ?e]
+                                                      [?p :block/children ?sib]
+                                                      [?sib :block/uid ?sib-uid]
+                                                      ;; filter selected
+                                                      [(= ?sib-uid ?selected)]
+                                                      [?sib :block/order ?o]]
+                                                    db
+                                                    (last uids))
+                                               (sort-by second))
+        [uid order]                       (last selected-sibs-of-last)
+        parent-eid                        (:db/id (common-db/get-parent db [:block/uid uid]))
+        n                                 (count selected-sibs-of-last)
+        reindex                           (common-db/minus-after db
+                                                                 parent-eid
+                                                                 order
+                                                                 n)
+        retracted-vec                     (mapcat #(common-db/retract-uid-recursively-tx db %) uids)
```

Comment:
Obs: I'm never really sure in cases like this if it's nicer to do `#(common-db/retract-uid-recursively-tx db %)` or `(partial common-db/retract-uid-recursively-tx db)`. 

I kinda prefer using `partial` because it conveys the meaning better (I'm setting the first param). But it's more chars (`partial` vs `# %`), `partial` doesn't work when I need to do other param manipulation, and using the fn as a param itself might mess with the editor tooling.

Kinda wish there was a more elegant way of doing param manipulation.

Reply:
I actually did not know about `partial`  I used `# %` while learning clojure and solving 4clojure problems, also did not see it in the code I read. Thanks to you I learned a  new thing, I agree to need a more elegant way of doing param manipulation, also there are a few options to do the manipulation. It is good that there are options but then also need to take into consideration what to use, when to use.